BugsnagNetworkRequestPlugin/BugsnagNetworkRequestPlugin/BSGURLSessionTracingDelegate.m

```diff
@@ -32,34 +33,40 @@ - (BOOL)canTrace {
     return g_sink != nil;
 }
 
-static NSString *responseConclusions[10] = {
-    @"NSURLSession error",
-    @"NSURLSession succeeded",
-    @"NSURLSession succeeded",
-    @"NSURLSession succeeded",
-    @"NSURLSession failed",
-    @"NSURLSession error",
-    @"NSURLSession error",
-    @"NSURLSession error",
-    @"NSURLSession error",
-    @"NSURLSession error",
-};
-
-- (NSString *)conclusionForResponseCode:(NSInteger)responseCode {
-    return responseConclusions[(responseCode / 100) % 10];
++ (nonnull NSString *)messageForResponse:(NSHTTPURLResponse *)response {
+    if (response) {
+        if (100 <= response.statusCode && response.statusCode < 400) {
+            return @"NSURLSession succeeded";
+        }
+        if (400 <= response.statusCode && response.statusCode < 500) {
+            return @"NSURLSession failed";
+        }
+    }
+    return @"NSURLSession error";
 }
 
-static NSDictionary *_Nonnull queryItemsAsDict(NSArray<NSURLQueryItem *> *_Nullable queryItems) {
++ (nullable NSDictionary<NSString *, NSString *> *)urlParamsForQueryItems:(nullable NSArray<NSURLQueryItem *> *)queryItems {
+    if (!queryItems) {
+        return nil;
+    }
     NSMutableDictionary *result = [NSMutableDictionary new];
-    for(NSURLQueryItem *item in queryItems)
-    {
+    for (NSURLQueryItem *item in queryItems) {
         result[item.name] = item.value;
```

Comment:
If a key already exists, convert the value into an array of values. Something like:

```suggestion
        if ([result[item.name] isKindOfClass:[NSArray class]]) {
            [result[item.name] addObject:item.value];
        } else if (result[item.name]) {
            result[item.name] = @[result[item.name], item.value].mutableCopy;
        } else {
            result[item.name] = item.value;
        }
```

though its not exactly that simple, since [value can be nil](https://developer.apple.com/documentation/foundation/nsurlcomponents/1407752-queryitems):
> If a name-value pair has no equals sign, the value property of the corresponding query item is nil.

Reply:
Name-value pairs with no equals sign are represented as an `NSURLQueryItem` with `item.value = @""`